services/src/main/java/org/exoplatform/news/NewsServiceImpl.java

```diff
@@ -196,12 +196,40 @@ public NewsServiceImpl(RepositoryService repositoryService,
    * @throws Exception when error
    */
   public News createNews(News news) throws Exception {
+    SessionProvider sessionProvider = sessionProviderService.getSessionProvider(null);
+    Session session = sessionProvider.getSession(
+                                                 repositoryService.getCurrentRepository()
+                                                                  .getConfiguration()
+                                                                  .getDefaultWorkspaceName(),
+                                                 repositoryService.getCurrentRepository());
     if (StringUtils.isEmpty(news.getId())) {
       news = createNewsDraft(news);
     } else {
-      postNewsActivity(news);
+      postNewsActivity(news, session);
       updateNews(news);
-      sendNotification(news, NotificationConstants.NOTIFICATION_CONTEXT.POST_NEWS);
+      sendNotification(news, NotificationConstants.NOTIFICATION_CONTEXT.POST_NEWS, session);
+    }
+
+    if (news.isPinned()) {
+      pinNews(news.getId());
+    }
+    NewsUtils.broadcastEvent(NewsUtils.POST_NEWS, news.getId(), news.getAuthor());
+    return news;
+  }
+
+  public News createNews(News news, Session session) throws Exception {
+    if (StringUtils.isEmpty(news.getId())) {
+      news = createNewsDraft(news);
+    } else {
+      try {
+        postNewsActivity(news, session);
+        updateNews(news);
+        sendNotification(news, NotificationConstants.NOTIFICATION_CONTEXT.POST_NEWS, session);
+      } finally {
+        if (session != null) {
+          session.logout();
+        }
```

Comment:
Session should be closed in the same function where it was created.

Reply:
I forgot it,no need i will remove it ,